src/Middleware/OutputCaching/src/OutputCacheKeyProvider.cs

```diff
@@ -29,173 +30,262 @@ internal OutputCacheKeyProvider(ObjectPoolProvider poolProvider, IOptions<Output
         _options = options.Value;
     }
 
-    // GET<delimiter>SCHEME<delimiter>HOST:PORT/PATHBASE/PATH<delimiter>H<delimiter>HeaderName=HeaderValue<delimiter>Q<delimiter>QueryName=QueryValue1<subdelimiter>QueryValue2<delimiter>R<delimiter>RouteName1=RouteValue1<subdelimiter>RouteName2=RouteValue2
+    // <VaryByKeyPrefix><delimiter>GET<delimiter>SCHEME<delimiter>HOST:PORT/PATHBASE/PATH<delimiter>H<delimiter>HeaderName=HeaderValue<delimiter>Q<delimiter>QueryName=QueryValue1<subdelimiter>QueryValue2<delimiter>R<delimiter>RouteName1=RouteValue1<subdelimiter>RouteName2=RouteValue2
```

Comment:
nit: like mentioned in some meeting, add a newline in the comment so it's readable in the editor